tutorials/build_systems/spack/spack_test.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2016-2021 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+@rfm.simple_test
+class BZip2SpackCheck(rfm.RegressionTest):
+    descr = 'Demo test using Spack to build the test code'
+    valid_systems = ['*']
+    valid_prog_environs = ['builtin']
+    executable = 'bzip2'
+    executable_opts = ['--help']
+
+    @run_before('compile')
+    def setup_build_system(self):
+        self.build_system = 'Spack'
```

Comment:
How about moving the `build_system` assignment to the class body and renaming this hook to `set_spack_environment`?